### PR TITLE
add better retry limit and recovery function for connect job

### PR DESCRIPTION
**Pull request by anandkumarpatel**
- set unlimited limit when connecting to swarm
- emit dock disconnect if limit reached for dock connect

### CODE REVIEW

File: lib/workers/docker.events-stream.connect.js [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+const sinceMap = require('../since-map')
+const Swarm = require('../swarm')
 
 module.exports.jobSchema = schemas.dockerEventsStreamConnect
 
+module.exports.maxNumRetries = 9
+
+module.exports.retryDelay = 1000
+
+// if we can not connect, we must destory
```
Comment (podviaznikov) [inline lib/workers/docker.events-stream.connect.js]: destory

File: lib/workers/docker.events-stream.connect.js [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
+module.exports.finalRetryFn = (job) => {
```
Comment (thejsj) [inline lib/workers/docker.events-stream.connect.js]: Any reason for not doing `Promise.method`?

**Review by thejsj — APPROVED** [review]
One nit. LGTM

**Comment by anandkumarpatel** [comment]
Last login: Thu Sep 15 16:37:52 on ttys001
docks-cli$ c snoop-a-loop
snoop-a-loop$ ssh gamma-dock-services sudo docker ps | grep docker
c8e37027f000        registry.runnable.com/runnable/docker-listener:tune-limits   "/bin/sh -c 'ulimit -"   3 hours ago         Up 3 hours                                   romantic_bell
snoop-a-loop$ npm start -- --gamma --timeout 50000

> snoop-a-loop@1.0.0 start /Users/anandkumarpatel/run/snoop-a-loop
> ./node_modules/mocha/bin/mocha -t 10000 -b index.js "--gamma" "--timeout" "50000"

  Cleanup
    ✓ should fetch the instances (839ms)
    ✓ should delete/destroy the non-repo container (194ms)
    ✓ should delete/destroy the repo container (301ms)
1. New Service Containers
   Creating Container
     ✓ should fetch all template containers (778ms)
     ✓ should copy the source instance (2026ms)
     ✓ should create the build (186ms)
     ✓ should build the build (809ms)
     ✓ should create an instance (519ms)
   Working Container
     ✓ should have a dockerContainer (3020ms)
     ✓ should get logs for that container (83ms)
     ✓ should be succsefully built (1010ms)
     ✓ should have a working terminal (2013ms)
2. New Repository Containers
   Create A Container
     Github
       ✓ should create a github org
       ✓ should fetch a github branch (499ms)
       ✓ should fetch a github repo branch (492ms)
     Source Context
       ✓ should fetch the source context (185ms)
       ✓ should fetch the source context versions (201ms)
     Context & Context Versions
       ✓ should create a context (195ms)
       ✓ should create a context version (1119ms)
       ✓ should fetch the stack analysis (1095ms)
       ✓ should copy the files (2631ms)
       ✓ should create an AppCodeVersion (1585ms)
     Builds & Instances
       ✓ should create a build for a context version (670ms)
       ✓ should build the build (281ms)
       ✓ should create an instance (543ms)
   Working Container
     ✓ should have a dockerContainer (7052ms)
     ✓ should get logs for that container (226ms)
     ✓ should be successfully built (1012ms)
     ✓ should have a working terminal (1151ms)
3. New Repository Containers created using a mirrored docker file
   Create A Container
     Github
       ✓ should create a github org
       ✓ should fetch a github branch (509ms)
       ✓ should fetch a github repo branch (521ms)
     Source Context
       ✓ should fetch the source context (182ms)
       ✓ should fetch the source context versions (181ms)
     Context & Context Versions
       ✓ should create a context (220ms)
       ✓ should create a context version (924ms)
       ✓ should update the context version when the dockerfile is mirrored (469ms)
       ✓ should fetch the stack analysis (203ms)
       ✓ should copy the files (2584ms)
       ✓ should create an AppCodeVersion (1319ms)
     Builds & Instances
       ✓ should create a build for a context version (401ms)
       ✓ should build the build (254ms)
       ✓ should create an instance (809ms)
   Working Container
     ✓ should have a dockerContainer (3019ms)
     ✓ should get logs for that container (194ms)
     ✓ should be successfully built (1513ms)
     ✓ should have a working terminal (1159ms)
     ✓ should reflect the mirrored dockerfile configuration (1138ms)
4. Rebuild Repo Container
   Rebuilding without Cache
     ✓ should deep copy the build (539ms)
     ✓ should rebuild the instance without cache (724ms)
   Working Container
     ✓ should have a container (7064ms)
     ✓ should get logs for that container (430ms)
     ✓ should be succsefully built (1008ms)
     ✓ should have a working terminal (337ms)
5. Github Webhooks
   Creating Branch Container
     ✓ should created a new branch (913ms)
     ✓ should create a new instance with the branch name (6001ms)
   Working Container
     ✓ should have a container (5038ms)
     ✓ should get logs for that container (538ms)
     ✓ should be succsefully built (1006ms)
     ✓ should have a working terminal (1822ms)
6. Isolation
   Create Container To Isolate
     Create A Container
       Github
         - should create a github org
         - should fetch a github branch
         - should fetch a github repo branch
       Source Context
         - should fetch the source context
         - should fetch the source context versions
       Context & Context Versions
         - should create a context
         - should create a context version
         - should fetch the stack analysis
         - should copy the files
         - should create an AppCodeVersion
       Builds & Instances
         - should create a build for a context version
         - should build the build
         - should create an instance
     Working Container
       - should have a dockerContainer
       - should get logs for that container
       - should be successfully built
       - should have a working terminal
   Create Isolation
   - should create the isolation
   - should create the instances for that isolation
     Isolated Service Container
   
   ```
   - should have a dockerContainer
   - should get logs for that container
   - should be successfully built
   - should have a working terminal
   ```
   
     Isolated Repo Container
       - should have a dockerContainer
       - should get logs for that container
       - should be successfully built
       - should have a working terminal
7. Container To Container DNS
   Repo Instance
     ✓ should connect to the container from the master branch (1340ms)
     ✓ should connect to the container from the newly created branch (if not isolated) (731ms)
   - should connect to the isolated container from the isolated branch
     Service Instance
   - should connect to the master branch repo instance
   - should connect to the creaated branch repo instance
8. Navi URLs
   Repo Instance
     ✓ should access the main container (274ms)
     ✓ should access the branch container (160ms)
   Service Instance
   - should connect to the service instance
   
   64 passing (1m)
   31 pending

snoop-a-loop$

File: lib/workers/docker.events-stream.connect.js (hunk 1 above)
Comment (anandkumarpatel) [reply lib/workers/docker.events-stream.connect.js]: de-story, its the end of their story 😄

**Pull request merged**